projects/telemetry/src/nmea_checksum.c

```diff
@@ -0,0 +1,79 @@
+#include "nmea_checksum.h"
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+// Private method to convert hex char to int
+static uint8_t prv_hex_to_int(char h) {
+  if ('0' <= h && h <= '9') {
+    return h - '0';
+  } else if ('A' <= h && h <= 'F') {
+    // We only care about uppercase because the NMEA messages are all caps
+    return h - 'A' + 10;
+  }
+  // Just to make the checksum fail
+  return 16;
+}
+
+static StatusCode prv_checksum_to_int(char tens, char ones, uint8_t *computed) {
+  uint8_t int_tens = prv_hex_to_int(tens);
+  uint8_t int_ones = prv_hex_to_int(ones);
+
+  // Checks for the 16 because it is the result sent in case of error
+  if (ones == 16 || tens == 16) {
+    return STATUS_CODE_INVALID_ARGS;
+  }
+  if (computed != NULL) {
+    *computed = int_tens * 16 + int_ones;
```

Comment:
Just so you know, this is equivalent to `((x & 0xF) << 4) | (y & 0xF)`. (masks are mostly for safety)